satellite/analytics/service.go

```diff
@@ -27,12 +38,15 @@ const (
 type Config struct {
 	SegmentWriteKey string `help:"segment write key" default:""`
 	Enabled         bool   `help:"enable analytics reporting" default:"false"`
+	HubSpotAPIKey   string `help:"hubspot api key" default:""`
 }
 
 // Service for sending analytics.
 //
 // architecture: Service
 type Service struct {
+	ctx           context.Context
+	cancel        context.CancelFunc
```

Comment:
to fix the errors in your build, I think you can add
```
type Service struct {
    ...
    startChan      chan struct{}
```
and inside `NewService`, add `startChan: make(chan struct{}),`
inside `Run`, add 
```
	service.ctx = ctx
	service.cancel = cancel
        close(service.startChan)
```

Finally, you can add `<- service.startChan` before functionality that requires `service.ctx`. For example
```
go func() {
    <- service.startChan
    err := service.sendHubSpotEvent(service.ctx, &dataEvent)
    ...
```